core/templates/dev/head/pages/story_editor/StoryEditorStateServiceSpec.js

```diff
@@ -0,0 +1,290 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for StoryEditorStateService.
+ */
+
+describe('Story editor state service', function() {
+  var StoryEditorStateService = null;
+  var StoryObjectFactory = null;
+  var StoryUpdateService = null;
+  var fakeEditableStoryBackendApiService = null;
+
+  var FakeEditableStoryBackendApiService = function() {
+    var self = {};
+
+    var _fetchOrUpdateStory = function() {
+      return $q(function(resolve, reject) {
+        if (!self.failure) {
+          resolve(self.newBackendStoryObject);
+        } else {
+          reject();
+        }
+      });
+    };
+
+    self.newBackendStoryObject = {};
+    self.failure = null;
+    self.fetchStory = _fetchOrUpdateStory;
+    self.updateStory = _fetchOrUpdateStory;
+    return self;
+  };
+
+  beforeEach(module('oppia'));
+  beforeEach(module('oppia', GLOBALS.TRANSLATOR_PROVIDER_FOR_TESTS));
+  beforeEach(module('oppia', function($provide) {
+    fakeEditableStoryBackendApiService = (
+      new FakeEditableStoryBackendApiService());
+    $provide.value(
+      'EditableStoryBackendApiService',
+      [fakeEditableStoryBackendApiService][0]);
+  }));
+
+  beforeEach(inject(function($injector) {
+    StoryEditorStateService = $injector.get(
+      'StoryEditorStateService');
+    StoryObjectFactory = $injector.get('StoryObjectFactory');
+    StoryUpdateService = $injector.get('StoryUpdateService');
+    $q = $injector.get('$q');
+    $rootScope = $injector.get('$rootScope');
+    $scope = $rootScope.$new();
+
+    fakeEditableStoryBackendApiService.newBackendStoryObject = {
+      id: '0',
+      title: 'Story title',
+      description: 'Story Description',
+      notes: '<p>Notes/p>',
+      story_contents: {
+        initial_node_id: 'node_1',
+        next_node_id: 'node_2',
+        nodes: []
+      },
+      language_code: 'en',
+      schema_version: '1',
+      version: '1'
+    };
+
+    secondBackendStoryObject = {
+      id: '0',
+      title: 'Story title  2',
+      description: 'Story Description 2',
+      notes: '<p>Notes 2/p>',
+      story_contents: {
+        initial_node_id: 'node_2',
+        next_node_id: 'node_1',
+        nodes: []
+      },
+      language_code: 'en',
+      schema_version: '1',
+      version: '1'
+    };
+  }));
+
+  it('should request to load the story from the backend', function() {
+    spyOn(
+      fakeEditableStoryBackendApiService, 'fetchStory').and.callThrough();
+
+    StoryEditorStateService.loadStory(0, 5);
```

Comment:
Shouldn't the arguments be strings?

Also consider making them more semantic, e.g. 'topicId0', 'storyId5'

Reply:
Done.